alerting/alerter-war/src/main/java/org/hawkular/metrics/alerter/ConditionManager.java

```diff
@@ -489,15 +495,17 @@ private void evaluateEach(String target, Map<String, Map<String, List<? extends
             }
         }
 
-        private void evaluate(String target, Map<String, BucketPoint> queryResults, ConditionEvaluator evaluator) {
+        private void evaluate(String target, Map<String, BucketPoint> queryResults, ConditionExpression expression) {
+            ConditionEvaluator evaluator = expression.getEvaluator();
             Map<String, String> preparedCondition = null;
             try {
                 preparedCondition = evaluator.prepare(queryResults);
             } catch (Exception e) {
                 log.debugf("Could not prepare evaluation, Skipping due to [%s]", e.getMessage());
             }
 
-            if (evaluator.evaluate()) {
+            boolean evalResult = evaluator.evaluate();
+            if (!isQuiet("_ALL_", expression.getQuietCount(), evalResult)) {
```

Comment:
I see your point, but I guess for now I'd prefer to leave things unchanged.  By using a special metricName the underlying mechanism can just use it as a Map key and not have to know it's something special.